### PR TITLE
Add boilerplate for CAPH test automation.

### PR DESCRIPTION
**Pull request by cwize1**
This change is progress towards creating test automation
where CAPH can be tested running against a mock version
of cloud agent.

Changes include:

1. Modify the Makefile and co. to allow publishing container image to
custom container registry.

2. Create scripts for setting up k3d cluster with CAPI/CAPH. These will
be used by the tests but they are also handy for manual testing
(especially if you want to avoid Kind to avoid the requirement to disable
the swap file on your dev box).

3. Create helper library/package for creating a mock cloud agent gRPC
server.

4. Create a mockcloudagent executable that reports an error for all requests.
This can be useful for manual testing. Though in the future, it could be
extended with more interesting logic.

5. Create the boilerplate for the tests, which simply sets up a mock cloud agent
and a CAPI/CAPH deployment (using k3d) and ensures CAPH can communicate with
the mock cloud agent.

### CODE REVIEW

**Comment by cwize1** [comment]
By the way, this change

**Comment by cwize1** [comment]
FYI: This change demonstrates how to test K8s controllers without needing to use a public container registry.

File: test/functional/Scripts/CAPISetup.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    echo "$(echo -n "$1" | base64 | tr -d '\n')"
+}
+ 
+export AZURESTACKHCI_CLOUDAGENT_FQDN_B64="$(to_base64 $AZURESTACKHCI_CLOUDAGENT_FQDN)"
+export WSSD_DEBUG_MODE_B64="$(to_base64 $WSSD_DEBUG_MODE)"
+export AZURESTACKHCI_BINARY_LOCATION_B64="$(to_base64 $AZURESTACKHCI_BINARY_LOCATION)"
+
+# Initialize CAPI and CAPH.
+clusterctl init --infrastructure azurestackhci --bootstrap kubeadm:v0.3.5 --control-plane kubeadm:v0.3.5 --core cluster-api:v0.3.5
+
```
Comment (zawachte-msft) [inline test/functional/Scripts/CAPISetup.sh:26]: can we put this all in the hack directory?

**Comment by zawachte-msft** [comment]
> FYI: This change demonstrates how to test K8s controllers without needing to use a public container registry.

You can also use tilt. https://github.com/tilt-dev/tilt

**Pull request closed without merging**